src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs

```diff
@@ -725,8 +718,7 @@ internal void FinishOperationSuccess(SocketError socketError, int bytesTransferr
                     // Mark socket connected.
                     if (socketError == SocketError.Success)
                     {
-                        if (s_loggingEnabled)
-                            SocketsEventSource.Connected(_currentSocket, _currentSocket.LocalEndPoint, _currentSocket.RemoteEndPoint);
+                        NetEventSource.Connected(_currentSocket, _currentSocket.LocalEndPoint, _currentSocket.RemoteEndPoint);
```

Comment:
> what's the general principle?

For me the general principle was: is there non-trivial work happening at the call site... if yes, it must be guarded, if no, it doesn't need to be.

We could say that every single call site to any tracing method in NetEventSource needs to have an IsEnabled check before it.  It'll lead to clunkier code, but it'll be easy to explain.  Is that what you'd prefer?